File: flutter/lib/ui/editor/time_format.dart
/// Timeline clock, as mm:ss.
///
/// Fixed width on purpose: a readout that switches between "9.4s" and "1:09"
/// changes length as it counts, which makes the surrounding row twitch while
/// the video plays.
String formatTime(int us) {
  final totalSeconds = (us < 0 ? 0 : us) ~/ 1000000;
  final minutes = (totalSeconds ~/ 60).toString().padLeft(2, '0');
  final seconds = (totalSeconds % 60).toString().padLeft(2, '0');
  return '$minutes:$seconds';
}

/// The same clock with tenths, for the playhead and for the short durations in
/// the panels, where whole seconds would hide the value being dragged.
String formatTimePrecise(int us) {
  final safe = us < 0 ? 0 : us;
  final totalSeconds = safe ~/ 1000000;
  final minutes = (totalSeconds ~/ 60).toString().padLeft(2, '0');
  final seconds = (totalSeconds % 60).toString().padLeft(2, '0');
  final tenths = (safe % 1000000) ~/ 100000;
  return '$minutes:$seconds.$tenths';
}

File: android/app/src/main/java/com/vixel/studio/ui/video/TimeFormat.kt
package com.vixel.studio.ui.video

/**
 * Timeline clock, as mm:ss.
 *
 * Fixed width on purpose: a readout that switches between "9.4s" and "1:09"
 * changes length as it counts, which makes the surrounding row twitch while
 * the video plays.
 */
fun formatTime(us: Long): String {
    val totalSeconds = (us.coerceAtLeast(0L)) / 1_000_000
    return "%02d:%02d".format(totalSeconds / 60, totalSeconds % 60)
}

/** Same clock with tenths, for the playhead, where a frame's worth matters. */
fun formatTimePrecise(us: Long): String {
    val safe = us.coerceAtLeast(0L)
    val totalSeconds = safe / 1_000_000
    val tenths = (safe % 1_000_000) / 100_000
    return "%02d:%02d.%d".format(totalSeconds / 60, totalSeconds % 60, tenths)
}
